### PR TITLE
UHF-9791: UHF-9791: Refactor javascript code to be more readable

### PR DESCRIPTION
**Pull request by Arkkimaagi**
# [UHF-9791](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9791)

* Code needed some refactoring

## What was done

* Code was refactored

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-9791_clean_code`
* Run `make drush-cr`

## How to test

* [ ] Check that cookie banner works as it should
* [ ] Check that code follows our standards

## Designers review

* [X] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

[UHF-9791]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/js/hds-cc/hds-cc.js [hunk 1]
```diff
@@ -1,41 +1,173 @@
...
+    const cookieSettingsGroups = [...cookieSettings.requiredCookies.groups, ...cookieSettings.optionalCookies.groups];
+
+    // Checksums for all groups calculated in parallel without waiting for each
+    await Promise.all(cookieSettingsGroups.map(async (group) => {
+      group.checksum = await this.#getChecksum(group);
+    }));
 
-    // Debug helper. Open banner and run on console to see the box updated.
-    window.aaa_chatcheck = () => dispatchEvent(new CustomEvent('CONSENTS_CHANGED', { detail: { groups: ['chat'] } }));
+    // Save chat group to accepted groups and update checkbox state
```
Comment (j-mys) [inline src/js/hds-cc/hds-cc.js:156]: By use case this is chat group, but syntax wise should be 'acceptedGroupsArray'

File: src/js/hds-cc/hds-cc.js [hunk 2]
```diff
@@ -48,66 +180,118 @@ class HdsCc {
...
     }
+  }
+
+
 
-    return cookie;
+  /**
+   * Removes the banner and related elements.
+   * @private
+   * @memberof ClassName
```
Comment (j-mys) [inline src/js/hds-cc/hds-cc.js:208]: Might be unnecessary definition

File: src/js/hds-cc/hds-cc.js [hunk 3]
```diff
@@ -48,66 +180,118 @@ class HdsCc {
...
-    cookieSettings.requiredCookies.groups.forEach(group => {
-      if (acceptedGroupNames.includes(group.groupId)) {
-        acceptedGroups[group.groupId] = group.checksum;
-      }
-    });
 
-    // Add accepted group names to acceptedGroups, assume others to be declined (overrides previous selections)
-    cookieSettings.optionalCookies.groups.forEach(group => {
+    // Find group checksums for accepted groups
+    const bothGroups = [...cookieSettings.requiredCookies.groups, ...cookieSettings.optionalCookies.groups];
```
Comment (j-mys) [inline src/js/hds-cc/hds-cc.js:254]: Could be more accurate as 'allGroups' or 'bothCategories'

**Review by j-mys — APPROVED** [review]
Nice improvements

File: src/js/hds-cc/hds-cc.js (hunk 1 above)
Comment (Arkkimaagi) [reply src/js/hds-cc/hds-cc.js]: Fixed

File: src/js/hds-cc/hds-cc.js (hunk 2 above)
Comment (Arkkimaagi) [reply src/js/hds-cc/hds-cc.js]: Removed

File: src/js/hds-cc/hds-cc.js (hunk 3 above)
Comment (Arkkimaagi) [reply src/js/hds-cc/hds-cc.js]: Changed to allGroups

**Pull request merged**